src/Raven.Studio/typescript/viewmodels/database/documents/editDocument.ts

```diff
@@ -1120,20 +1120,26 @@ class editDocument extends shardViewModelBase {
             })
             .fail((xhr: JQueryXHR) => {
                 // if revisions is enabled try to load revisions bin entry
-                if (xhr.status === 404 && db.hasRevisionsConfiguration()) {
-                    this.loadRevisionsBinEntry(id)
-                        .done(doc => {
-                            if (doc) {
-                                loadTask.resolve(doc);
-                            } else {
-                                messagePublisher.reportWarning("Could not find document: " + id);
-                                loadTask.reject();
-                            }
-                        })
-                        .fail(() => loadTask.reject());
+                if (xhr.status === 404) {
+                    if (db.hasRevisionsConfiguration()) {
+                        this.loadRevisionsBinEntry(id)
+                            .done(doc => {
+                                if (doc) {
+                                    loadTask.resolve(doc);
+                                } else {
+                                    messagePublisher.reportWarning("Could not find document: " + id);
+                                    loadTask.reject();
+                                }
+                            })
+                            .fail(() => loadTask.reject());
+                    } else {
+                        this.dirtyFlag().reset();
+                        messagePublisher.reportWarning("Could not find document: " + id);
+                        loadTask.reject();
+                    }
                 } else {
                     this.dirtyFlag().reset();
-                    messagePublisher.reportWarning("Could not find document: " + id);
+                    messagePublisher.reportError("Could not load document: " + id);
```

Comment:
I think we could at least state the status code and status message?
It would say "server unavailable" which would be a valuable message.

Reply:
Ping @ml054

Reply:
@ml054 ping again

Reply:
done